### ISSUE TITLE
[Request] Making "Official" spectator slots eligible for Reserved Slots

### ISSUE THREAD

**Issue by TA-Red** [Person8880/Shine#656]
I'm not honestly sure if it's possible, but I was wondering if an addition to the reserved slot plugin could be put in. Specifically: allowing shine to designate some of the "official spectator slots" as reserved slots.

Currently if someone adds reserved slots to the server they're counted as playable slots. So as an example, my current server is 24 active players, 4 official spectator slots. If I were to add four reserved slots to a server they would only apply to active player slots. So if no reserved players are on the server it would default to a 10v10 game, which is not something I want. I want to maintain a player count of 12v12 active players. I would like reserved players to be automatically sent to the official spectate slots so that when a slot opens up on 12v12, they can join.

I'm not honestly sure if this is possible with the current official spectator slots, but I thought I'd ask anyway.

So basically long story short, I just want to be able to make "official spectator slots" into reserved slots.

If clarification is needed, just ask away. I'm not entirely sure I explained what I was requesting very well, just did my best.

**Comment by Person8880** [comment]
What should it do if a player without reserved access joins into a playable slot but then goes into spectate? This would normally switch them to a spectator slot if one is free.

Should it prevent the normal behaviour, and leave all spectator slots free for players with reserved access only? Or should it switch their slot type and compensate by reserving the now unoccupied player slot?

**Comment by TA-Red** [comment]
It may be easier to paint the picture of how I'd use this system in the real world scenario. My goal is to have 24 player slots, and 8 spectator slots. Of those 8 slots, I'd like 4 to be set as reserved slots.

So currently if the server is completely full and is set to 24 active players and 4 spectator slots and an active player joins spectate, that player isn't counted as a spectator slot until one of the current spectators joins the game. Realistically that's how it would behave.

So for example if I had 24 active players, and 8 spectator slots with 4 of those slots being for reserved. Assume for a moment that there are no reserved players in the game. If one of those active players went spectate it would again, not count them as a spectator until one of the current spectators joined the game. It would treat it the same as if there were four reserved players actively there.

So basically: Any additional spectators past the spec slot limit are not counted as spectators until a current spectator joins the game. Which, to my understanding, is how it's treated now.

Again though, this is why I wasn't sure if this was even possible. But I figured it couldn't hurt to ask.

**Comment by Person8880** [comment]
The next update will add the `SlotType` option. This can be one of:
* `PLAYABLE` - the current behaviour, reserve only player slots.
* `ALL` - your desired behaviour, reserve based on all slots, starting with spectator slots (basically counting total connected clients vs. max players + max spectators - reserved slots).

You'll want to set `SlotType` to `"ALL"` and the number of slots to 4. This will reserve 4/8 of your spectator slots, and none of the player slots.
